server/on-demand-entry-handler.js

```diff
@@ -0,0 +1,221 @@
+import DynamicEntryPlugin from 'webpack/lib/DynamicEntryPlugin'
+import { EventEmitter } from 'events'
+import { createServer } from 'http'
+import { join } from 'path'
+import resolvePath from './resolve'
+import touch from 'touch'
+import WebSocket from 'ws'
+
+const ADDED = Symbol()
+const BUILDING = Symbol()
+const BUILT = Symbol()
+
+export default async function onDemandEntryHandler (devMiddleware, compiler, {
+  dir,
+  dev,
+  maxInactiveAge = 1000 * 25
+}) {
+  const server = await getServer()
```

Comment:
Here we are starting a websocket server to accept pings.
The best idea is to hook into our existing app. 
But with our API, we don't have access to the server object. So, we can't do that without changing the API.

Since this is dev only, it's fine to start a new server. Which is lightweight of course.

### Why?

Earlier we did by sending a HTTP request from the client for every 5 secs. But those requests could be blocked at any time by the chrome for sending too much of requests.

And that could block HMR updates as well.
So, that's why we moved to WS.